com.microsoft.mrtk.input/Interactors/InteractorVisuals/MRTKBaseReticleVisual.cs

```diff
@@ -0,0 +1,72 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.XR.Interaction.Toolkit;
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    [DisallowMultipleComponent]
+    public class MRTKBaseReticleVisual : MonoBehaviour, IXRCustomReticleProvider
```

Comment:
Needs documentation/comments... not entirely sure what this is for, if MRTKRayInteractorVisual isn't inheriting from it?

Reply:
MRTKRayInteractor visual should inherit from this. The separation of responsibilities is

base: Takes care of swapping reticle visuals by implementing `IXRCustomReticleProvider`
RayVisuals: Places the reticle at the end of the ray, orients it, etc.

Reply:
Alrighty, sounds good... perfect for a doccomment :D

Reply:
Done